apps/web/pages/api/integrations.ts

```diff
@@ -30,7 +30,17 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   }
 
   if (req.method == "DELETE") {
-    const id = req.body.id;
+    /** Get the id from the stripe_payment credential for this user */
+    const paymentCredential = await prisma.credential.findFirst({
+      where: {
+        userId: session?.user?.id,
+        type: "stripe_payment",
+      },
+      select: {
+        id: true,
+      },
+    });
```

Comment:
As we already have credentialId we can use findFirst with where: { id: req.body.id }
```suggestion
    /** Get the id from the stripe_payment credential for this user */
    const stripePaymentCredential = await prisma.credential.findFirst({
      where: {
        id: req.body.id,
        userId: session?.user?.id,
        type: "stripe_payment",
      },
      select: {
        id: true,
      },
    });
```